### PR TITLE
Fixes for CSV Download issues

### LINKED ISSUE #16: Update column headings
Display a shorter version of the column heading, with a longer version becoming visible on hover.

See "Metalation calculator column labels.docx" in shared drive.

Consider using [MathJax](https://www.mathjax.org) to improve appearance of symbols, or see https://sites.psu.edu/symbolcodes/accents/math/#browser for other ideas.

Ensure the CSV export uses plain text headings.

### PR DESCRIPTION
**Pull request by zeynepaki**
* Fixes #16 

- [ ] Write a clean data method
- [ ] rewrite the CSV download method
- [ ] Add descriptions of headings to the civ file in `# ..` comment format
- [ ] Fix the issue of element symbols being displayed as on

### CODE REVIEW

File: typescript/src/main.ts [hunk 1]
```diff
@@ -178,35 +178,72 @@ function reset() {
...
       row.push('"' + data + '"');
     }
     csv.push(row.join(separator));
   }
+  const explanation = [];
+  const headings = rows[0].cells;
+  for (let k = 0; k < headings.length; k++) {
+    const spans = headings[k].getElementsByTagName("span");
+    if (spans.length > 0) {
+      let span = spans[0].innerHTML;
```
Comment (alisonrclarke) [inline typescript/src/main.ts:237]: Just a small thing but it might be better to name this e.g. `spanText` or `detailText` to show that it's a string rather than an HTML Element. Ditto with header.

**Review by alisonrclarke — COMMENTED** [review]
Code looks great, just some comments/variable naming could be a bit clearer.

File: typescript/src/main.ts [hunk 2]
```diff
@@ -178,35 +178,72 @@ function reset() {
...
+      const inputs = [];
+      if (unfilteredInputs.length > 0) {
+        for (let l = 0; l < unfilteredInputs.length; l++) {
+          const input = unfilteredInputs[l];
+          if (input.type == "number") {
+            inputs.push(input);
+          }
+        }
+      }
+
```
[the remark this answers is not in the document]
Comment (zeynepaki) [reply typescript/src/main.ts:219]: I could feel there must be a better way of doing this just didn't know how! Learnt something new today!

File: typescript/src/main.ts (hunk 1 above)
Comment (zeynepaki) [reply typescript/src/main.ts:237]: Pardon my lack of slang but does ditto mean its alright or change that as well?

Comment (alisonrclarke) [reply typescript/src/main.ts:237]: Sorry! ditto means the same applies to, so please change header too.

File: typescript/src/main.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (zeynepaki) [reply typescript/src/main.ts:219]: Screenshot 2022-01-17 at 13 35 04

[the remark this answers is not in the document]
Comment (alisonrclarke) [reply typescript/src/main.ts:225]: The line I gave was probably JS from testing on the console rather than typescript - sorry!

Try something like:

`Array.from(document.getElementsByTagName('input')).filter(e => e.type == 'checkbox')`

File: typescript/src/main.ts [hunk 3]
```diff
@@ -178,35 +178,75 @@ function reset() {
...
       }
-      // Remove line breaks and escape double-quote with double-double-quote
-      data = data.replace(/(\r\n|\n|\r)/gm, "").replace(/(\s\s)/gm, " ");
-      data = data.replace(/"/g, '""');
-      data = data.replace(/\u2206/g, "Delta ");
+      /**
+      Text is "cleaned up" to be more readable
+      and the delta symbol is replaced with the word "Delta"
+      as excel does not display unicode symbols correctly.
+      **/
```
Comment (alisonrclarke) [inline typescript/src/main.ts:228]: Could you move this to a comment on the `cleanData` function?

**Review by alisonrclarke — COMMENTED** [review]
Just a couple of comment changes requested!

File: typescript/src/main.ts [hunk 4]
```diff
@@ -212,17 +212,10 @@ function downloadTableAsCsv(tableId: string, separator = ",") {
...
-      if (unfilteredInputs.length > 0) {
-        for (let l = 0; l < unfilteredInputs.length; l++) {
-          const input = unfilteredInputs[l];
-          if (input.type == "number") {
-            inputs.push(input);
-          }
-        }
-      }
-
+      //const unfilteredInputs = cols[j].getElementsByTagName("input");
```
[the remark this answers is not in the document]
Comment (zeynepaki) [reply typescript/src/main.ts]: I forgot to press save, realised it when I was trying to change directories

**Review by alisonrclarke — APPROVED** [review]

**Review by alisonrclarke — APPROVED** [review]
